docs: add sections to tell usage of Draft PR (#4677)
Kong/kubernetes-ingress-controller: CONTRIBUTING.md

- bug fix pull requests should be opened against `main` as the base branch
- feature pull requests should be opened with `next` as the base branch

> If your pull request is still in progress, but you want get some early reviews before

Reviewer: ```suggestion
If your pull request is still in progress, but you want to get an earl review before
```

Reviewer: Nit: I believe this should be placed in its own section, not in `Finding work`. Something like `Working with Pull Requests` maybe.

Author: updated.

Reviewer: ping: https://github.com/Kong/kubernetes-ingress-controller/pull/4677#discussion_r1329856858

Author: ADDED the section and a simple guide of submitting PRs.

- bug fix pull requests should be opened against `main` as the base branch
- feature pull requests should be opened with `next` as the base branch

## Working with Pull Requests 

When you finish your work and want your contributions to be included, please submit a pull request
and include necessary information about the pull request in its description following our
[pull request template](https://github.com/Kong/kubernetes-ingress-controller/blob/a5cfca4a97927c2878fce593929a8e48442f3127/.github/PULL_REQUEST_TEMPLATE.md). 
A clear title and description is helpful to encourage your pull request to be reviewed and merged.

If your pull request is still in progress, but you want to get early reviews before
it is ready to be merged, please add `do not merge` or `work in progress` labels to the
pull request. If your pull request is not ready to be reviewed yet, please convert it to `Draft`.

> `Draft` pull requests will not be reviewed until they are converted to `Open` status.

Reviewer 2: ```suggestion
`Draft` pull requests will not be reviewed until they are converted to `Open` status.

```

- bug fix pull requests should be opened against `main` as the base branch
- feature pull requests should be opened with `next` as the base branch

## Working with Pull Requests 

When you finish your work and want your contributions to be included, please submit a pull request
and include necessary information about the pull request in its description following our
[pull request template](https://github.com/Kong/kubernetes-ingress-controller/blob/a5cfca4a97927c2878fce593929a8e48442f3127/.github/PULL_REQUEST_TEMPLATE.md). 
A clear title and description is helpful to encourage your pull request to be reviewed and merged.

If your pull request is still in progress, but you want to get early reviews before
it is ready to be merged, please add `do not merge` or `work in progress` labels to the
pull request. If your pull request is not ready to be reviewed yet, please convert it to `Draft`.
`Draft` pull requests will not be reviewed until they are converted to `Open` status.

> 
> 
> 

Reviewer: ```suggestion

```
